### ISSUE TITLE
Ability to specify module-level values as volatile/not-Final

### ISSUE THREAD

**Issue by PeterJCLaw** [python/mypy#9209]
Not sure if this is a bug or expected - module level values are treated as constants, I'd like a way to indicate that they're expected to vary.

For example, given a module like this:
``` python
# debug.py
enable_warning: bool = False  # behaves the same whether annotated or not

def warning(message):
    if enable_warning:
        print(message)
```

Which is used like this:
``` python
from . import debug

def set_level(warning: bool):
    debug.enable_warning = warning

debug.warning('foo')
```

Currently with `warn_unreachable` enabled, mypy will complain that the `print` in `debug.py` isn't reachable. While I can see why that is seen to be the case from just looking at `debug.py`, it would be great to be able to mark "configuration" values like this as expecting to change, so that they're treated more generally as `bool` than `Literal[False]`.

I'm using mypy 0.782 on Python 3.6.

**Comment by Akuli** [comment]
I can't reproduce this.

```
$ git checkout v0.782
HEAD is now at 39b6bc47 Bump version to 0.782
$ head bar/*
==> bar/debug.py <==
enable_warning: bool = False
def warning(message):
    if enable_warning:
        print(message)

==> bar/foo.py <==
from . import debug

def set_level(warning: bool):
    debug.enable_warning = warning

debug.warning("foo")

==> bar/__init__.py <==
$ python3.6 -m mypy bar/*
Success: no issues found in 3 source files
```

Which mypy flags are you using?

**Comment by PeterJCLaw** [comment]
Ah, apologies, it seems I made an assumption when cutting this down. The original has a `None` value which should actually be `Optional[...]`, which I missed. The error I was getting was about an `and` condition being partially redundant and I think I just misread what was happening. If I annotate the thing which was `func = None` but which should be `func: Optional[...] = None` then that fixes it.
